### PR TITLE
Address several old issus higher-order FEM, poisson tests, and testing problems

### PR DESCRIPTION
**Pull request by cekees**
# Mandatory Checklist

Please ensure that the following criteria are met:

- [x] Title of pull request describes the changes/features
- [x] Request at least 2 reviewers
- [x] If new files are being added, the files are no larger than 100kB. Post the file sizes.
- [x] Code coverage did not decrease. If this is a bug fix, a test should cover that bug fix. If a new feature is added, a test should be made to cover that feature.
- [x] New features have appropriate documentation strings (readable by sphinx)
- [x] Contributor has read and agreed with CONTRIBUTING.md and has added themselves to CONTRIBUTORS.md 

As a general rule of thumb, try to follow [PEP8](https://www.python.org/dev/peps/pep-0008/) guidelines.

# Description

Initial work to close out #14, #15, and #16. Mainly fixing the tests themselves and petsc options. In the process I found that the DG archiving had never been properly updated and that a larger DG boundary penalty improves conditioning.  There were some messy test interactions that caused other changes--a project for the future.  On a positive note, @ejtovar @jhcollins @zhang-alvin I used the pytest parameterized testing support to generate more comprehensives tests here: https://github.com/erdc/proteus/pull/1206/files?file-filters%5B%5D=.csv&file-filters%5B%5D=.py#diff-50950c09443ffea929f7eb626c9cb336. It's nothing new but might be a useful example for you when writing tests. It replaces the old "yield tests".

### CODE REVIEW

File: proteus/NumericalFlux.py [hunk 1]
```diff
@@ -761,7 +761,7 @@ def __init__(self,vt,getPointwiseBoundaryConditions,
         Advection_DiagonalUpwind_Diffusion_IIPG.__init__(self,vt,getPointwiseBoundaryConditions,
                                                          getAdvectiveFluxBoundaryConditions,
                                                          getDiffusiveFluxBoundaryConditions)
-        self.penalty_constant = 10.0
+        self.penalty_constant = 100.0
```
Comment (zhang-alvin) [inline proteus/NumericalFlux.py:764]: Might be good to make a global constant again.

File: proteus/TwoPhaseFlow/utils/Parameters.py [hunk 2]
```diff
@@ -224,7 +224,7 @@ def __init__(self,
         # self.n.linearSmoother = None
         # NUMERICAL FLUX
         # self.n.massLumping = False
-        # self.n.conservativeFlux = None
+        self.n.conservativeFlux = None
```
Comment (ejtovar) [inline proteus/TwoPhaseFlow/utils/Parameters.py:227]: What does un-commenting this do?

**Comment by cekees** [comment]
> I saw that there were slight changes in how the petsc options were being loaded into the tests, or at least the defaults were reset. Does that resolve #1184 on an individual test level?

No, not really. I think that issue needs a more comprehensive review of how they're being set/loaded and a good across-the-board cleanup of that issue.

File: proteus/TwoPhaseFlow/utils/Parameters.py (hunk 2 above)
Comment (cekees) [reply proteus/TwoPhaseFlow/utils/Parameters.py:227]: None is the default, and it should be None here, so it's really just explicitly setting it to the default. There seem to be some strange test interactions with pytest, however, and I had an issue with this being set to something besides None.

File: proteus/NumericalFlux.py (hunk 1 above)
Comment (cekees) [reply proteus/NumericalFlux.py:764]: On this one I'm going to just leave it as is. The best value of the penalty_constant in the DG-methods varies, unfortunately.

**Pull request merged**